includes/amp-helper-functions.php

```diff
@@ -1603,8 +1604,8 @@ function amp_get_content_sanitizers( $post = null ) {
 	 */
 	$sanitizers['AMP_Style_Sanitizer']['allow_transient_caching'] = apply_filters( 'amp_parsed_css_transient_caching_allowed', true );
 
-	// Force style sanitizer, meta sanitizer, and validating sanitizer to be at end.
-	foreach ( [ 'AMP_Style_Sanitizer', 'AMP_Meta_Sanitizer', 'AMP_Tag_And_Attribute_Sanitizer' ] as $class_name ) {
+	// Force style, meta, layout and validating sanitizers to be at the end.
+	foreach ( [ 'AMP_Style_Sanitizer', 'AMP_Meta_Sanitizer', 'AMP_Layout_Sanitizer', 'AMP_Tag_And_Attribute_Sanitizer' ] as $class_name ) {
```

Comment:
I think that `AMP_Layout_Sanitizer` has to come before `AMP_Style_Sanitizer`. Otherwise, the `style` attribute will always be removed at this point and this condition will never be `true`:

https://github.com/ampproject/amp-wp/blob/c6aed8f66921a527911fa74188bf838a3d164f56/includes/sanitizers/class-amp-layout-sanitizer.php#L55

Reply:
```suggestion
	foreach ( [ 'AMP_Layout_Sanitizer', 'AMP_Style_Sanitizer', 'AMP_Meta_Sanitizer', 'AMP_Tag_And_Attribute_Sanitizer' ] as $class_name ) {
```

Reply:
Oh right yes that makes sense.

Reply:
I'm having trouble reproducing a problem caused by this, however.

Reply:
Here, this markup causes the problem:

```html
<div style="position:relative; height: 400px; width:400px"><amp-img src="https://amp-wp.org/wp-content/uploads/2020/08/Asset-33@2x.png" style="width:100%;height:100%;top:0;left:0;bottom:0;right:0;background:blue; outline:solid 1px lime !important;"></amp-img></div>
```

Without putting the layout sanitizer first, I get an error:

![image](https://user-images.githubusercontent.com/134745/93278931-d59b0a80-f77a-11ea-9553-e7fe77853bc8.png)